### PR TITLE
use declarative jenkinsfile syntax

### PR DESCRIPTION
**Pull request by camallen**
switch the jenkinsfile to match other repo jenkinsfile syntax and workflow.

Some good examples of the differences between scripted (current) and declarative (this PR) syntax - https://e.printstacktrace.blog/jenkins-scripted-pipeline-vs-declarative-pipeline-the-4-practical-differences/

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -1,28 +1,72 @@
...
     stage('Build Docker image') {
-        newImage = docker.build(dockerImageName)
-        newImage.push()
-        newImage.push(scmVars.GIT_COMMIT)
+      agent any
+      steps {
+        script {
+          def dockerRepoName = 'zooniverse/aggregation-for-caesar'
+          def dockerImageName = "${dockerRepoName}:${GIT_COMMIT}"
+          def buildArgs = "--build-arg REVISION='${GIT_COMMIT}' ."
```
Comment (CKrawczyk) [inline Jenkinsfile:17]: Just looking at the other PR, does this need to use `scmVars.GIT_COMMIT` rather than `GIT_COMMIT`?

**Review by CKrawczyk — COMMENTED** [review]
Unsure if the `GIT_COMMIT` should be `scmVars.GIT_COMMIT` like in the other PR?

File: Jenkinsfile (hunk 1 above)
Comment (camallen) [reply Jenkinsfile:17]: no - in the declarative version these values are available as ENV vars directly, not from an explicit checkout var result. e.g. https://github.com/zooniverse/caesar/blob/bf65f2c62f29f1f4c397d7adb3646f76582959ec/Jenkinsfile#L17

**Review by CKrawczyk — APPROVED** [review]
I am not an expert on jenkinsfiles but all seems to be in order.

**Pull request merged**